src/logic/calculate.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-unused-vars */
+import operate from './operate';
+/* eslint-enable no-unused-vars */
+
+const calculate = (calcData, buttonName) => {
+  let { next, total, operation } = calcData;
+
+  switch (buttonName) {
+    case '+/-':
+      total *= -1;
+      next *= -1;
+      break;
+    case 'AC':
+      total = null;
+      next = null;
+      break;
+    case '%':
+      operation = '%';
+      break;
+    case '.':
+      operation = '.';
+      break;
+    case '+':
+      operation = '+';
+      break;
+    case '-':
+      operation = '-';
+      break;
+    case '/':
+      operation = '/';
+      break;
+    case '*':
+      operation = '*';
+      break;
+    case '=':
+      operation = '=';
+      break;
+    case '1':
+      next = 1;
+      break;
+    case '2':
+      next = 2;
+      break;
+    case '3':
+      next = 3;
+      break;
+    case '4':
+      next = 4;
+      break;
+    case '5':
+      next = 5;
+      break;
+    case '6':
+      next = 6;
+      break;
+    case '7':
+      next = 7;
+      break;
+    case '8':
+      next = 8;
+      break;
+    case '9':
+      next = 9;
+      break;
+    case '0':
+      next = 0;
```

Comment:
### [Optional]

Your code appears to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code

We could improve this code by using [.include](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

Example: 
```
  const number = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
  case number.includes(buttonName):
```